debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/antlr/listener/ColumnDefinitionParserListener.java

```diff
@@ -214,14 +214,14 @@ else if (dataTypeContext instanceof MySqlParser.CollectionDataTypeContext) {
             }
         }
 
-        String dataTypeName = dataType.name();
+        String dataTypeName = dataType.name().toUpperCase();
 
         if (dataTypeName.equals("ENUM") || dataTypeName.equals("SET")) {
             // type expression has to be set, because the value converter needs to know the enum or set options
-            columnEditor.type(dataTypeName.toUpperCase(), getText(dataTypeContext));
+            columnEditor.type(dataTypeName, getText(dataTypeContext));
```

Comment:
The problem was exactly here - when the type was not `ENUM` but `enum` we did not get to this line and thus there were no enum options  parsed. So the easiest way how to reproduce the issue was to send an SQL statement with `ENUM` not in uppercase.